### DISCUSSION TITLE
Data list - Umbraco Dictionary Items - keword instead empty option when no translation

### DISCUSSION THREAD

**Discussion by tomotrepo** [leekelleher/umbraco-contentment#378] [Idea]
Hi, for the 'Data list' element and the 'Umbraco Dictionary Items' data source, it would be good if the editor could at least see the keyword if, for example, it has no translations entered. After all, it is the editor's view. There is also nothing wrong if the keyword appears on the frontend as well (although this is a matter of implementation), as the editor knows that it is missing and will enter it. We use this method often as it serves us as a quick codebook for a wide variety of things. But it is disadvantageous if, for example, you forget to enter the translation for certain keyword, then in the administration it appears, for example, dropdown with empty options. Regards

**Comment by leekelleher** [comment]
@tomotrepo Thank you for the suggestion.

I'm wondering what can be amended in the data-source's code to resolve this? 🤔 
https://github.com/leekelleher/umbraco-contentment/blob/4.6.0/src/Umbraco.Community.Contentment/DataEditors/DataList/DataSources/UmbracoDictionaryDataListSource.cs#L89-L98

**Reply by tomotrepo** [reply]
If anything helps :), this is from my custom dictionary-picker plugin back from v7

```
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

using Umbraco.Web.WebApi;
using Umbraco.Web.Editors;
using Umbraco.Core.Persistence;

namespace A8
{
    [Umbraco.Web.Mvc.PluginController("A8")]
    public class A8DictionaryPickerApiController : UmbracoAuthorizedJsonController
    {
        public IEnumerable<DictionaryKey> GetValues()
        {
            var requestedParentDictionaryKeyword = A8.Common.CheckStr(HttpContext.Current.Request.QueryString["requestedParentDictionaryKeyword"]);
            if (requestedParentDictionaryKeyword.Length > 0)
            {
                var db = UmbracoContext.Application.DatabaseContext.Database;
                return db.Fetch<DictionaryKey>("select a.[key] as [Key], coalesce(cast((select top 1 b.[value] from cmsLanguageText b where b.UniqueId = a.id and b.languageId = 2) as nvarchar(max)) + ' ', '') + '(' + a.[key] + ')' AS [Value] from cmsDictionary as a left outer join cmsDictionary b on a.parent = b.id where b.[key] = '" + A8.Common.StringToSql(requestedParentDictionaryKeyword) + "' order by a.[key]");
            }
            return null;
        }
    }

    public class DictionaryKey
    {
		public string Key { get; set; }
		public string Value { get; set; }
   }
}
```

'b.languageId = 2' is of course default language and 'requestedParentDictionaryKeyword' is input parameter

Regards
